### PR TITLE
fix #18, better package init.

### LINKED ISSUE #18: Version number in package.json templates can be unexpected for new projects
If I understand the flow correctly, we have two different forces at play here...

I use the **shield-study-cli** to create a new shield study, which does a `git clone mozilla/shield-studies-addon-template` to create the scaffolding.

One of the problems is that I'd expect my package.json versioning for a new Shield study to start at a semver logical "0.1.0" or even "1.0.0", but the version number will actually be the semver of the published npm module **shield-studies-addon-template** (which is currently 0.0.2).

Similarly, if we assume that somebody wouldn't want to deploy their shield study to npm, we'd probably want to put a `"private": true` in the package.json, but then that also implies that it'd go in the **shield-studies-addon-template/package.json**. Which I _suppose_ is OK since we may not be publishing to npm if we're relying on installing it via `git clone` versus `npm install` as a dependency somewhere.

### PR DESCRIPTION
**Pull request by gregglind**
Solutions proposed
- shield init blah => directory of shield-study-blah
- package.json gets modified (some)

### CODE REVIEW

File: scripts/alter-package-json.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ modify mozilla/shield-studies-addon-template/package.json in place
+ as part of `shield init`
+*/
+
+var jsonPath  = process.argv[2];
+var dir = process.argv[3];
+
+console.log(jsonPath, dir);
```
Comment (pdehaan) [inline scripts/alter-package-json.js]: Remove the `console.log()`

File: scripts/alter-package-json.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+var p = require(jsonPath);
+
+var oldName = 'shield-studies-addon-template';
+var newName = ['shield-study-',''][~~dir.startsWith('shield-study-')]+dir;
```
Comment (pdehaan) [inline scripts/alter-package-json.js]: I had to read this a few times to get the "gist" of it. Not sure if we can simplify or refactor this slightly.

This seems to give me similar results on my basic JSBin testing:

``` js
const newName = dir.startsWith('shield-study-') ? dir : `shield-study-${dir}`;
```

Comment (pdehaan) [reply scripts/alter-package-json.js]: Also, this seems duplicated from [index.js:193](https://github.com/mozilla/shield-study-cli/pull/19/files#diff-168726dbe96b3ce427e7fedce31bb0bcR193) above. Not sure if we want to use some shared function somewhere versus duplicating the default naming logic.

File: scripts/initStudy.bash [hunk 3]
```diff
@@ -23,6 +26,7 @@ echo "
 ## Next Steps
 
 - cd '$1'
+- npm install --progress
```
Comment (pdehaan) [inline scripts/initStudy.bash:29]: Should we do this automatically?

**Comment by pdehaan** [comment]
r+.
Tested locally and it worked well!

File: scripts/alter-package-json.js (hunk 1 above)
Comment (gregglind) [reply scripts/alter-package-json.js]: OK :)

File: scripts/alter-package-json.js (hunk 2 above)
Comment (gregglind) [reply scripts/alter-package-json.js]: If it happens again, will turn it into "a thing"

File: scripts/initStudy.bash (hunk 3 above)
Comment (gregglind) [reply scripts/initStudy.bash:29]: I tried to from the shell script, using npm install --no-progress.  Something about how stdout gets managed made it seem weird and hanging.  Opening as a new bug. #20

**Pull request merged**